src/plugins/intel_cpu/src/nodes/mkldnn_rnn.cpp

```diff
@@ -600,107 +583,122 @@ void MKLDNNRNN::copyWeightsData() {
     if (cell_type == mkldnn::algorithm::vanilla_lstm) {
         gate_map = gate_map_lstm;
         if (G > gate_map_lstm_size) {
-            IE_THROW() << "G isn't equal to the size of gate_map";
+            THROW_ERROR << ". G isn't equal to the size of gate_map.";
         }
     } else if (cell_type == mkldnn::algorithm::vanilla_gru) {
         gate_map = gate_map_gru;
         if (G > gate_map_gru_size) {
-            IE_THROW() << "G isn't equal to the size of gate_map";
+            THROW_ERROR << ". G isn't equal to the size of gate_map";
         }
     } else if (cell_type == mkldnn::algorithm::lbr_gru) {
         gate_map = gate_map_gru;
         if (G > gate_map_gru_size) {
-            IE_THROW() << "G isn't equal to the size of gate_map";
+            THROW_ERROR << ". G isn't equal to the size of gate_map.";
         }
     } else if (cell_type == mkldnn::algorithm::vanilla_rnn) {
         gate_map = gate_map_rnn;
         if (G > gate_map_rnn_size) {
-            IE_THROW() << "G isn't equal to the size of gate_map";
+            THROW_ERROR << ". G isn't equal to the size of gate_map.";
         }
     } else {
         gate_map = gate_map_gru;
         if (G > gate_map_gru_size) {
-            IE_THROW() << "G isn't equal to the size of gate_map";
+            THROW_ERROR << ". G isn't equal to the size of gate_map.";
         }
     }
 
     if (runtimePrecision == Precision::BF16) {
         fillWeights<uint16_t>(gate_map, wIdx, rIdx);
     } else if (runtimePrecision == Precision::FP32) {
         // WA To avoid different weights layer and iter formats in FP32 case
-        if (T != 1 || N < 16)
-            w_format = mkldnn::memory::format_tag::ldigo;
+        if ((T.isStatic() && T.maxVal != 1) || (N.isStatic() && N.maxVal < optimalBatchSize))
```

Comment:
Shouldn't we handle dynamic cases as well? Like if T.minVal > 1 or N.maxVal < 16 the condition is still valid.

Reply:
Modified